### PR TITLE
Data extraction - Pokesniper for rare Pokemon

### PR DESCRIPTION
**Pull request by vivek-sethia**
Data Extraction implemented for getting rare Pokemon appearance details. 
1. First phase listening through api and storing the data into temporary json files( npm run listen to execute)
2. Second phase storing the data into the database from temporary files and deleting them.
   ( npm run update --collection=rarePokemon to execute )

### CODE REVIEW

File: app/stores/rarePokemon.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+     */
+    add: function (data, callback) {
+        var rarePokemons = new rarePokemon();
+
+        // traversing through the data
+        for (var key in data) {
+            if (data.hasOwnProperty(key)) {
+               // processing the data for coordinates since they are not separated as latitude
+               // and longitudes in the original data source
+                if( key == 'coords'){
```
Comment (swathi-ssunder) [inline app/stores/rarePokemon.js]: Use ===

File: services.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+    getHttpRequestData : function(url){
+        var request = require('request');
+        var fs = require('fs');
+        var all = [];
+
+        request(url, function (error, response, body) {
+
+            if (!error && response.statusCode == 200) {
+                var data = JSON.parse(body);
```
Comment (swathi-ssunder) [inline services.js]: Use ===

**Comment by jonas-he** [comment]
:+1:

**Comment by vivek-sethia** [comment]
Suggested changes done and changed listener to generic functionality 
Now listener would be executed with this command npm run listen --collection=<collection_name>
E.g. npm run listen --collection=rarePokemon

**Pull request closed without merging**